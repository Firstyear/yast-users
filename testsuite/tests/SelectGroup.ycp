{

    include "testsuite.ycp";
    import "Users";

    Users::groups ["local" ] = $[
        100: $[ "groupname": "users", "userlist":"", "more_users": "hh","type": "local" ],
        17:  $[ "groupname": "audio", "userlist":"hh", "more_users": "","type": "local" ]
    ];

    Users::groups ["ldap" ] = $[
        100: $[ "groupname": "users", "userlist":"", "more_users": "u1,u2","type": "ldap" ]
    ];

    DUMP (sformat ("local groups::\n %1", Users::groups ["local"]:$[]));

    DUMP (sformat ("ldap groups::\n %1", Users::groups ["ldap"]:$[]));

    DUMP ("==========================================================");

    Users::current_groups = ["local", "ldap"];

    integer gid = 17;

    TEST(``(Users::SelectGroup (gid)), [], 0);

    DUMP (sformat ("selected group with id %1:\n %2",gid,Users::group_in_work));

    DUMP ("==========================================================");

    gid = 100;

    TEST(``(Users::SelectGroup (gid)), [], 0);

    DUMP (sformat ("selected group with id %1:\n %2",gid,Users::group_in_work));


}
