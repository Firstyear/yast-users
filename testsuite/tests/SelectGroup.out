Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Dump	local groups::
Dump	 $[17:$["groupname":"audio", "more_users":"", "type":"local", "userlist":"hh"], 100:$["groupname":"users", "more_users":"hh", "type":"local", "userlist":""]]
Dump	ldap groups::
Dump	 $[100:$["groupname":"users", "more_users":"u1,u2", "type":"ldap", "userlist":""]]
Dump	==========================================================
Return	nil
Dump	selected group with id 17:
Dump	 $["groupname":"audio", "more_users":"", "type":"local", "userlist":"hh"]
Dump	==========================================================
Return	nil
Dump	selected group with id 100:
Dump	 $["groupname":"users", "more_users":"hh", "type":"local", "userlist":""]
